File: app/src/main/java/net/clipcodes/myapplication/ui/activities/SplashscreenActivity.java
package net.clipcodes.myapplication.ui.activities;

import android.app.Activity;
import android.content.Intent;
import android.graphics.PixelFormat;
import android.os.Bundle;
import android.util.Log;
import android.view.Window;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.ImageView;
import android.widget.LinearLayout;

import com.kakao.auth.Session;
import com.kakao.network.ErrorResult;
import com.kakao.usermgmt.UserManagement;
import com.kakao.usermgmt.callback.UnLinkResponseCallback;

import net.clipcodes.myapplication.R;
import net.clipcodes.myapplication.accounts.SessionCallback;

public class SplashscreenActivity extends Activity {
    private String TAG = "SplashscreenActivity";

    public void onAttachedToWindow() {
        super.onAttachedToWindow();
        Window window = getWindow();
        window.setFormat(PixelFormat.RGBA_8888);
    }
    /** Called when the activity is first created. */
    Thread splashTread;
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splashscreen);
        checkKakaoLoginSession();
//        onClickLogout();
        StartAnimations();
    }

    private void checkKakaoLoginSession(){
        if(Session.getCurrentSession().checkAndImplicitOpen()){
            Log.e(TAG, "토큰큰 : " + Session.getCurrentSession().getTokenInfo().getAccessToken());
            Log.e(TAG, "토큰큰 리프레쉬토큰 : " + Session.getCurrentSession().getTokenInfo().getRefreshToken());
            Log.e(TAG, "토큰큰 파이어데이트 : " + Session.getCurrentSession().getTokenInfo().getRemainingExpireTime());

            Session.getCurrentSession().addCallback(SessionCallback.getInstance());
            SessionCallback.getInstance().requestMe();
        }
    }

    private void StartAnimations() {
        Animation anim = AnimationUtils.loadAnimation(this, R.anim.anim_splash_alpha_color);
        anim.reset();
        LinearLayout l=(LinearLayout) findViewById(R.id.lin_lay);
        l.clearAnimation();
        l.startAnimation(anim);

        anim = AnimationUtils.loadAnimation(this, R.anim.anim_translate);
        anim.reset();
        ImageView iv = (ImageView) findViewById(R.id.splash);
        iv.clearAnimation();
        iv.startAnimation(anim);

        splashTread = new Thread() {
            @Override
            public void run() {
                try {
                    int waited = 0;
                    // Splash screen pause time
                    while (waited < 2000) {
                        sleep(100);
                        waited += 100;
                    }
                    Intent intent = new Intent(SplashscreenActivity.this,
                            MainActivity.class);
                    intent.setFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION);
                    startActivity(intent);
                    SplashscreenActivity.this.finish();
                } catch (InterruptedException e) {
                    // do nothing
                } finally {
                    SplashscreenActivity.this.finish();
                }

            }
        };
        splashTread.start();

    }
}

File: ui/src/main/java/kr/co/geniemarket/ui/SplashscreenActivity.kt
package kr.co.geniemarket.ui

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.ViewModelStoreOwner
import kr.co.geniemarket.core.GMLog
import kr.co.geniemarket.ui.databinding.ActivitySplashscreenBinding

@SuppressLint("CustomSplashScreen")
class SplashscreenActivity : AppCompatActivity() {

    companion object {
        private val TAG : String = SplashscreenActivity::class.java.simpleName
    }

    private var mBinding: ActivitySplashscreenBinding? = null

    private val binding get() = mBinding!!

    private lateinit var viewModelProvider : ViewModelProvider

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        mBinding = ActivitySplashscreenBinding.inflate(layoutInflater)
        setContentView(binding.root)

        GMLog.setLogLevel(GMLog.LogLevel.Verbose)
        GMLog.i("GenieMarket version: ${BuildConfig.GENIEMARKET_VERSION}")

        viewModelProvider = ViewModelProvider(
            this as ViewModelStoreOwner,
            ViewModelProvider.NewInstanceFactory())

        startAnimation()
    }

    private fun startAnimation() {
        var anim : Animation = AnimationUtils.loadAnimation(this, R.anim.anim_splash_alpha_color)
        anim.reset()
        binding.linLay.clearAnimation()
        binding.linLay.startAnimation(anim)

        anim = AnimationUtils.loadAnimation(this, R.anim.anim_translate)
        anim.reset()
        binding.splash.clearAnimation()
        binding.splash.startAnimation(anim)

        Thread() {
            try {
                var waited = 0
                while (waited < 2000) {
                    Thread.sleep(100)
                    waited += 100
                }
                val intent = Intent(this, MainActivity::class.java)
                intent.setFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION)
                startActivity(intent)
                this@SplashscreenActivity.finish()
            }catch (ex : Exception) {
                GMLog.e(ex.toString(), ex)
            }finally {
                this@SplashscreenActivity.finish()
            }
        }.start()
    }
}